test/trustedresources.go

```diff
@@ -324,34 +370,68 @@ func signInterface(signer signature.Signer, i interface{}) ([]byte, error) {
 	return sig, nil
 }
 
-// GetSignedV1beta1Pipeline signed the given pipeline and rename it with given name
-func GetSignedV1beta1Pipeline(unsigned *v1beta1.Pipeline, signer signature.Signer, name string) (*v1beta1.Pipeline, error) {
-	signedPipeline := unsigned.DeepCopy()
-	signedPipeline.Name = name
-	if signedPipeline.Annotations == nil {
-		signedPipeline.Annotations = map[string]string{}
-	}
-	signature, err := signInterface(signer, signedPipeline)
-	if err != nil {
-		return nil, err
+// GetSignedPipeline signed the given pipeline and rename it with given name and apiVersion
+func GetSignedPipeline(unsigned runtime.Object, signer signature.Signer, name, apiVersion string) (runtime.Object, error) {
```

Comment:
one disadvantage of merging 2 helper functions is that we need to do type assert for the returned obj